lib/mintacoin/assets/crypto/stellar.ex

```diff
@@ -0,0 +1,86 @@
+defmodule Mintacoin.Assets.Stellar do
+  @moduledoc """
+  Implementation of the Stellar crypto functions for assets
+  """
+
+  alias Mintacoin.Assets.Crypto.AssetResponse
+  alias Stellar.{Horizon, Horizon.Transaction, KeyPair, TxBuild}
+
+  @type status :: :ok | :error
+  @type stellar_response :: map()
+  @type public_key :: String.t()
+  @type secret_key :: String.t()
+  @type error :: {:error, any()}
+
+  @behaviour Mintacoin.Assets.Crypto.Spec
+
+  @impl true
+  def create_asset(opts) do
+    [
+      distributor_secret_key: distributor_secret_key,
+      issuer_secret_key: issuer_secret_key,
+      asset_code: asset_code,
+      asset_supply: asset_supply
+    ] =
+      Keyword.take(opts, [:issuer_secret_key, :distributor_secret_key, :asset_code, :asset_supply])
+
+    {issuer_pk, _issuer_secret_key} = issuer_keypair = KeyPair.from_secret_seed(issuer_secret_key)
+
+    {distribution_pk, _sk} =
+      distribution_keypair = KeyPair.from_secret_seed(distributor_secret_key)
+
+    source_account = TxBuild.Account.new(issuer_pk)
+
+    {:ok, seq_num} = Horizon.Accounts.fetch_next_sequence_number(issuer_pk)
+    sequence_number = TxBuild.SequenceNumber.new(seq_num)
+
+    asset = [code: asset_code, issuer: issuer_pk]
+
+    trustline_operation =
+      TxBuild.ChangeTrust.new(
+        asset: asset,
+        source_account: distribution_pk
+      )
+
+    create_payment_operation =
+      TxBuild.Payment.new(
+        destination: distribution_pk,
+        asset: asset,
+        amount: asset_supply,
+        source_account: issuer_pk
+      )
+
+    issuer_signature = TxBuild.Signature.new(issuer_keypair)
+    distribution_signature = TxBuild.Signature.new(distribution_keypair)
+
+    {:ok, envelope} =
+      source_account
+      |> TxBuild.new(sequence_number: sequence_number)
+      |> TxBuild.add_operation(trustline_operation)
+      |> TxBuild.add_operation(create_payment_operation)
+      |> TxBuild.sign([issuer_signature, distribution_signature])
+      |> TxBuild.envelope()
+
+    envelope
+    |> Horizon.Transactions.create()
+    |> format_response()
+  end
+
+  @spec format_response({status(), stellar_response()}) :: {:ok, AssetResponse.t()} | error()
```

Comment:
@FelipeGuzmanSierra @jhonatan-kmt Please don't forget to add the params name for the types.
```suggestion
  @spec format_response(tx_response :: {status(), stellar_response()}) :: {:ok, AssetResponse.t()} | error()
```